apps/taskman/api.py

```diff
@@ -223,3 +223,29 @@ def put(self, request, task_key):
         ).update_task_status(
             issue_key=task_key, status=serializer.validated_data["status"]
         )
+
+
+@jira_token_description
+class task_assignee(GenericAPIView):
+    @extend_schema(
+        responses=JiraIssueQueryResultSerializer,
+    )
+    def get(self, request, user):
+        """Get a list of tasks from a user"""
+        return JiraTaskmanQuerier(
+            token=request.headers.get("JiraAuthentication")
+        ).search_tasks_by_assignee(user)
+
+    @extend_schema(
+        parameters=[
+            OpenApiParameter(name="task_key", required=True, type=str),
+        ],
+        description="Add a task into a group",
+        responses={204: OpenApiResponse(description="Modified.")},
+    )
+    def put(self, request, user):
+        """Assign a user for a task"""
+
+        return JiraTaskmanQuerier(
+            token=request.headers.get("JiraAuthentication")
+        ).assign_task(task_key=request.data["task_key"], assignee=user)
```

Comment:
This is a problem when you don't use serializers to validate input:

`task_key` is required but there is nothing enforcing that requirement, there is documentation (OpenAPI spec) that mentions that it is required, but what happens when someone calls this endpoint without a `task_key`?

They **should** get a 400 error explaining that `task_key` is a required attribute and that they did not provide it, what actually happens here is that you'll get a `KeyError` because there's no `task_key` in `request.data` and the server will return a cryptic error message with a 500 status which is not correct.

Please fix this and write a regression test, this potentially applies to the other views in taskman so I'd suggest checking for that too.

Reply:
makes sense, I will submit a fix for that.